curriculum/challenges/english/25-front-end-development/lecture-working-with-state-and-responding-to-events-in-react/67d1ebb595f4f619c0e35d1a.md

```diff
@@ -2,14 +2,165 @@
 id: 67d1ebb595f4f619c0e35d1a
 title: How Do You Update Objects in State?
 challengeType: 11
-videoId: nVAaxZ34khk
+videoId: BwHJrBW-m0M
 dashedName: how-do-you-update-objects-in-state
 ---
 
 # --description--
 
 Watch the lecture video and answer the questions below.
 
+# --transcript--
+
+How Do You Update Objects in State?
+
+Updating objects in state in React can be tricky if you're used to changing object property values directly.
+
+React treats state as immutable, meaning you should not modify it directly.
+
+Let's look at what happens if you try to change objects in React state directly, and then dive into the correct way to do it.
+
+Let's say you have an object in your component state that represents a user's profile, and you want the user to update their age: 
+
+```jsx
+import { useState } from "react";
+
+function Profile() {
+  const [user, setUser] = useState({
+    name: "John Doe",
+    age: 31,
+    city: "LA",
+  });
+
+  // Change user age directly
+  const handleAgeChange = (e) => {
+    user.age = e.target.value;
+    console.log(user);
+  };
+
+  return (
+    <div>
+      <h1>User Profile</h1>
+      <p>Name: {user.name}</p>
+      <p>Age: {user.age}</p>
+      <p>City: {user.city}</p>
+
+      <h2>Update User Age </h2>
+      <input type="number" value={user.age} onChange={handleAgeChange} />
+    </div>
+  );
+}
+
+export default Profile;
+```
+
+This code will not work because you're directly modifying the `user.age` property.
+
+Even though `console.log(user)` will show the new age in the console, React won't re-render the component to show it in the UI because you didn't use the setter function, `setUser`.
+
+To update an object directly in the state, you need to use the setter function to create a new object with the updated value. For example:
+
+```js
+const handleAgeChange = (e) => {
+  setUser({
+    age: e.target.value,
+  });
+};
+```
+
+That works. But if you look at the page now, the user's age gets updated, but the values for the name and city are lost.
+
+This is because the new object you passed to the setter function only contained a key/value pair for age.
+
+To prevent this from happening, you can copy the existing object first, then update only the property you want to update, in this case, age.
```

Comment:
```suggestion
To prevent this from happening, you can copy the existing object first, then update only the property you want to update, in this case, `age`.
```